### PR TITLE
Leave url field blank when creating cards from search page

### LINKED ISSUE #1558: Disable "url" Anki field when creating cards from search page.
The url field is very useful when sentence mining from websites but not at all for the search page. You end up with a useless wall of text on your Anki cards:
Screenshot_20241102_135703
I think it would be better if this field were left blank when the url starts with `chrome-extension` or `moz-extension`

### PR DESCRIPTION
**Pull request by dchen5022**
Fixes #1558 

Not sure how to add tests or if its necessary

### CODE REVIEW

**Comment by Kuuuube** [comment]
Does this work on edge? They have a different prefix registered but can't test this myself.

**Comment by jamesmaa** [comment]
Edge looks to be `extension` 
image

And also for e-readers/android the url is `kiwi-extension://`

**Comment by Kuuuube** [comment]
I'd recommend using a similar approach to this instead https://github.com/yomidevs/yomitan/blob/22da9dfa35227003ff934645f27941215e0b2bfb/ext/js/pages/settings/popup-preview-controller.js#L127.

**Comment by djahandarie** [comment]
Another option is to do something similar to the second half (after the &&) of this:
https://github.com/yomidevs/yomitan/blob/36221974bc09f6bdf94356718c47211c4544e022/ext/js/application.js#L199

Which checks if the URL matches the protocol of import.meta.url (which will be `chrome-extension` or whatever is appropriate for the given browser). This seems a bit better than having a big list of protocols maybe?

@Kuuuube Do you have a preference?

**Comment by dchen5022** [comment]
That looks like a cleaner solution

**Comment by Kuuuube** [comment]
> @Kuuuube Do you have a preference?

Didn't know that existed. Your idea sounds better.

**Comment by dchen5022** [comment]
Should I add a test case for this?

File: ext/js/data/anki-note-builder.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ export class AnkiNoteBuilder {
             if (internalFieldName === '{url}') {
                 const urlPrefix = '<a href="';
                 const url = value.replace(urlPrefix, '');
-                if (!['http:', 'https:', 'ws:', 'wss:', 'ftp:', 'data:', 'file:'].includes(new URL(url).protocol)) {
+                if (window.location.protocol === new URL(url).protocol) {
```
Comment (djahandarie) [inline ext/js/data/anki-note-builder.js:106]: I don't think this is quite right, as `window.location.protocol` can technically be `http(s)` if the code was to execute in a content script (though I guess this code wouldn't execute in a content script, it still doesn't feel right). You instead want `new URL(url).protocol === new URL(import.meta.url).protocol` I believe, as `import.meta.url` will always be the extension URL.

Comment (dchen5022) [reply ext/js/data/anki-note-builder.js:106]: Oh ok, thanks. I'll fix this then.

**Comment by dchen5022** [comment]
Thanks for the help! I'll fix that

**Review by Kuuuube — APPROVED** [review]

**Pull request merged**